src/main/java/com/fasterxml/jackson/core/util/BufferRecyclerPool.java

```diff
@@ -0,0 +1,17 @@
+package com.fasterxml.jackson.core.util;
+
+import com.fasterxml.jackson.core.TokenStreamFactory;
+
+/**
+ * Interface for entity that controls creation and possible reuse of {@link BufferRecycler}
+ * instances used for recycling of underlying input/output buffers.
+ *
+ * @since 2.16
+ */
+public interface BufferRecyclerPool
+    extends java.io.Serializable
+{
+    public BufferRecycler acquireBufferRecycler(TokenStreamFactory forFactory);
```

Comment:
@cowtowncoder I don't understand why this method takes a `TokenStreamFactory` as argument, can you please clarify?

Reply:
It allows inspecting configuration of the factory that creates parser/generator that needs recycler: mostly `JsonFactory.Feature` settings.
`TokenStreamFactory` is that abstract super-type of `JsonFactory`; could pass latter if that proves necessary.